catalog/spatial/registry/registry-api-impl/src/test/java/org/codice/ddf/registry/api/impl/TestRegistryStoreAutoPublisher.java

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.registry.api.impl;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.Optional;
+
+import org.codice.ddf.registry.api.RegistryStore;
+import org.codice.ddf.registry.common.metacard.RegistryObjectMetacardType;
+import org.codice.ddf.registry.federationadmin.service.FederationAdminService;
+import org.codice.ddf.registry.federationadmin.service.RegistryPublicationService;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.osgi.framework.ServiceReference;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.AttributeImpl;
+
+public class TestRegistryStoreAutoPublisher {
+
+    private RegistryStoreAutoPublisher registryStoreAutoPublisher;
+
+    private ServiceReference serviceReference;
+
+    @Test
+    public void testBindRegistryStoreNoContext() {
+        registryStoreAutoPublisher = spy(new RegistryStoreAutoPublisher());
+        serviceReference = mock(ServiceReference.class);
+        registryStoreAutoPublisher.bindRegistryStore(serviceReference);
+
+        verify(registryStoreAutoPublisher, Mockito.times(0)).registryPublish(any());
+    }
+
+    @Test
+    public void testRegistryPublishFailed() throws Exception {
+        registryStoreAutoPublisher = new RegistryStoreAutoPublisher();
```

Comment:
This block of code, here to line 64, could be refactored to a `setup()` method (not a `@BeforeTest`) that you invoke from this and the following test. You'd just have to add a few more fields to support that.

That would facilitate adding more methods.